stickermanager.py

```diff
@@ -0,0 +1,276 @@
+            'chat': event.chat_id,
+            'sender_id': event.sender_id,
+            'sender_name': sender_name,
+            'score': 0,
+            'emoji': emoji,
+            'votes': {},
+            'filepath': filename,
+            'pack_link': f'https://t.me/addstickers/{STICKER_PACK_SHORT_NAME}',
+            'animated': orig_evt.sticker and orig_evt.sticker.mime_type == 'application/x-tgsticker'
+        }
+        reply_evt: Message = await orig_evt.reply(
+            POLL_TEMPLATE.format_map(get_template_data()),
+            buttons=[Button.inline(UP, b'+'), Button.inline(DOWN, b'-')],
+            parse_mode='html')
+        current_vote['poll'] = reply_evt.id
+        asyncio.ensure_future(reply_evt.pin(notify=True), loop=bot.loop)
+
+    @bot.on(events.CallbackQuery(chats=ALLOWED_CHATS, data=lambda data: data in (b'+', b'-')))
+    async def vote_poll(event: events.CallbackQuery.Event) -> None:
+        if not current_vote or current_vote['poll'] != event.message_id:
+            await event.answer('That poll is closed.')
+            return
+
+        async with current_vote_lock:
+            await vote_poll_int(event)
+
+    async def vote_poll_int(event: events.CallbackQuery.Event) -> None:
+        global current_vote
+
+        if not current_vote or current_vote['poll'] != event.message_id:
+            await event.answer('That poll is closed.')
+            return
+
+        weight, displayname = WEIGHTS.get(event.sender_id, (DEFAULT_WEIGHT, None))
+        if weight == 0:
+            await event.answer('You don\'t have the permission to vote.')
+            return
+        if event.data == b'-':
+            weight = -weight
+
+        displayname = displayname or escape_html((await event.get_sender()).first_name)
+        try:
+            existing = current_vote['votes'][event.sender_id]
+            if existing.weight == weight:
+                await event.answer(f'You already voted {weight}')
+                return
+        except KeyError:
+            pass
+        current_vote['votes'][event.sender_id] = VoteData(weight=weight, displayname=displayname)
+
+        scores = calculate_scores()
+        current_vote['score'] = scores.sum
+
+        if abs(scores.sum) >= VOTES_REQUIRED:
+            if scores.sum > 0:
+                current_vote['result'] = RESULT_ADDED.format_map(get_template_data())
+                res = "accepted"
+            else:
+                current_vote['result'] = RESULT_REJECTED.format_map(get_template_data())
+                res = "rejected"
+            await bot.edit_message(current_vote['chat'], current_vote['poll'],
+                                   POLL_FINISHED_TEMPLATE.format_map(get_template_data()),
+                                   parse_mode='html')
+            await event.answer(f'Successfully voted {fancy_round(weight)},'
+                               f' which made the sticker be {res} \U0001f389')
+            if res == "accepted":
```

Comment:
Given `res` only seems to be used here, and checked against a value with two possible outcomes, wouldn't it be better to have `accepted = scores.sum > 0`?